src/map/clif.cpp

```diff
@@ -4490,91 +4490,72 @@ void clif_createchat( map_session_data& sd, e_create_chatroom flag ){
 	clif_send( &packet, sizeof( packet ), &sd.bl, SELF );
 }
 
-
-/// Display a chat above the owner (ZC_ROOM_NEWENTRY).
-/// 00d7 <packet len>.W <owner id>.L <char id>.L <limit>.W <users>.W <type>.B <title>.?B
+/// Calculates the chat status
 /// type:
 ///     0 = private (password protected)
 ///     1 = public
 ///     2 = arena (npc waiting room)
 ///     3 = PK zone (non-clickable)
-void clif_dispchat(struct chat_data* cd, int32 fd)
-{
-	unsigned char buf[128];
-	uint8 type;
+e_status_chatroom clif_chat_status( const chat_data &cd ){
+	if (cd.owner->type == BL_PC)
+		return (cd.pub) ? STATUSROOM_PUBLIC : STATUSROOM_PRIVATE;
+	else if (cd.owner->type == BL_NPC)
+		return (cd.limit) ? STATUSROOM_ARENA : STATUSROOM_PK_ZONE;
+	else
+		return STATUSROOM_PUBLIC;
+}
 
-	if( cd == nullptr || cd->owner == nullptr )
-		return;
+/// Calculates the chat user count
+uint16 clif_chat_usercount( const chat_data& cd ){
+	if( cd.owner->type == BL_NPC ){
+		return cd.users + 1;
+	}else{
+		return cd.users;
+	}
+}
+
+/// Display a chat above the owner.
+/// 00d7 <packet len>.W <owner id>.L <char id>.L <limit>.W <users>.W <type>.B <title>.?B (ZC_ROOM_NEWENTRY)
+void clif_dispchat( const chat_data& cd ){
+	PACKET_ZC_ROOM_NEWENTRY* p = reinterpret_cast<PACKET_ZC_ROOM_NEWENTRY*>( packet_buffer );
```

Comment:
A question after reading the check `if( cd == nullptr || cd->owner == nullptr )` (before the change)
`cd.owner` can't be nullptr anymore?